src/features/WriteStoriesModal/WriteStoriesModal.tsx

```diff
@@ -2,27 +2,272 @@ import React from "react";
 import { Button } from "../../components/Button/Button";
 
 import { Modal } from "../../components/Modal/Modal";
-import { StorybookLogo } from "../WelcomeModal/StorybookLogo";
-import { ModalContentWrapper } from "../WelcomeModal/WelcomeModal.styled";
+import { Icons } from "@storybook/components";
+import {
+  Description,
+  Header,
+  Image,
+  Main,
+  ModalContent,
+  SpanHighlight,
+} from "./WriteStoriesModal.styled";
+import { SyntaxHighlighter } from "../../components/SyntaxHighlighter/SyntaxHighlighter";
+import { List } from "../../components/List/List";
+import { ListItem } from "../../components/List/ListItem/ListItem";
+import { useGetButtonPath } from "./hooks/useGetButtonPath";
+import { useGetWarningButtonStatus } from "./hooks/useGetWarningButtonStatus";
+import { useGetBackdropBoundary } from "./hooks/useGetBackdropBoundary";
+import titleSidebarImg from "./assets/01-title-sidebar.png";
+import storyNameSidebarImg from "./assets/02-story-name-sidebar.png";
+import argsImg from "./assets/03-args.png";
+
+import dataJavascript from "./code/javascript";
+import dataTypescript from "./code/typescript";
+import dataTypescriptNextjs from "./code/nextjs-typescript";
+import { useGetProject } from "./hooks/useGetFrameworkName";
+import { API, AddonStore } from "@storybook/manager-api";
+
+export function WriteStoriesModal({
+  onFinish,
+  api,
+  addonsStore,
+}: {
+  onFinish: () => void;
+  api: API;
+  addonsStore: AddonStore;
+}) {
+  const [step, setStep] = React.useState<
+    "imports" | "meta" | "story" | "args" | "customStory"
+  >("imports");
+
+  const [isWarningStoryCopied, setWarningStoryCopied] = React.useState(false);
+
+  const buttonPath = useGetButtonPath();
+  const doesWarningButtonExist = useGetWarningButtonStatus(
```

Comment:
maybe this should be called `warningButtonStatus` instead? `doesWarningButtonExist` sounds like it refers to a boolean value

Reply:
I agree